packages/babel-traverse/src/path/context.ts

```diff
@@ -74,15 +82,17 @@ export function visit(this: NodePath): boolean {
     return false;
   }
 
-  // Note: We need to check "this.shouldSkip" twice because
-  // the visitor can set it to true. Usually .shouldSkip is false
+  const currentContext = this.context;
+  // Note: We need to check "this.shouldSkip" first because
+  // another visitor can set it to true. Usually .shouldSkip is false
   // before calling the enter visitor, but it can be true in case of
   // a requeued node (e.g. by .replaceWith()) that is then marked
   // with .skip().
-  if (this.shouldSkip || this.call("enter") || this.shouldSkip) {
+  if (this.shouldSkip || this.call("enter")) {
```

Comment:
If `this.enter("enter")` is `false`, either `fns` is empty (no visitors), or all the visitors does not change the traverse flags. If `fns` is empty, then `this.shouldSkip` would not be changed at all. If all the visitors do not change traversal flags, then `this.shouldSkip` should equals to `this.shouldSkip` before `enter` hook is called. Therefore the second `this.shouldSkip` check is redundant.